src/lib/seam/components/DeviceDetails/LockDeviceDetails.tsx

```diff
@@ -0,0 +1,158 @@
+      <ContentHeader title='Device' onBack={onBack} />
+      <div className='seam-body'>
+        <div className='seam-summary'>
+          <div className='seam-content'>
+            <div className='seam-image'>
+              <DeviceImage device={device} />
+            </div>
+            <div className='seam-info'>
+              <span className='seam-label'>{t.device}</span>
+              <h4 className='seam-device-name'>{device.properties.name}</h4>
+              <div className='seam-properties'>
+                <span className='seam-label'>{t.status}:</span>{' '}
+                <OnlineStatus device={device} />
+                <span className='seam-label'>{t.power}:</span>{' '}
+                <BatteryStatus device={device} />
+                <DeviceModel device={device} />
+              </div>
+            </div>
+          </div>
+          <Alerts alerts={alerts} className='seam-alerts-space-top' />
+        </div>
+        <div className='seam-box'>
+          <div
+            className='seam-content seam-access-codes'
+            onClick={toggleAccessCodesOpen}
+          >
+            <span className='seam-value'>
+              {accessCodeCount} {t.accessCodes}
+            </span>
+            <ChevronRightIcon />
+          </div>
+        </div>
+
+        <div className='seam-box'>
+          <div className='seam-content seam-lock-status'>
+            <div>
+              <span className='seam-label'>{t.lockStatus}</span>
+              <span className='seam-value'>{lockStatus}</span>
+            </div>
+            <div className='seam-right'>
+              <Button
+                size='small'
+                onClick={() => {
+                  toggleLock.mutate()
+                }}
+              >
+                {toggleLockLabel}
+              </Button>
+            </div>
+          </div>
+          <AccessCodeLength
+            supportedCodeLengths={
+              device.properties?.supported_code_lengths ?? []
+            }
+          />
+        </div>
+      </div>
+    </div>
+  )
+}
+
+function AccessCodeLength(props: {
+  supportedCodeLengths: number[]
+}): JSX.Element | null {
+  const { supportedCodeLengths } = props
+
+  if (supportedCodeLengths.length === 0) {
+    return null
+  }
+
+  const min = Math.min(...supportedCodeLengths)
+  const max = Math.max(...supportedCodeLengths)
+
+  const range = min === max ? max : `${min}–${max}`
+
+  return (
+    <div className='seam-content seam-access-code-length'>
+      <span className='seam-label'>{t.codeLength}</span>
+      <span className='seam-value'>
+        {range} {t.digits}
+      </span>
+    </div>
+  )
+}
+
+const t = {
+  device: 'Device',
+  unlock: 'Unlock',
+  lock: 'Lock',
+  locked: 'Locked',
+  unlocked: 'Unlocked',
+  accessCodes: 'access codes',
+  codeLength: 'Code length',
+  digits: 'digits',
+  lockStatus: 'Lock status',
+  status: 'Status',
+  power: 'Power',
+}
```

Comment:
:penguin: 

ha pseudo-i18n, fwiw I don't love the enum-oriented pattern because I think it makes contribution harder (project search often begins with a string seen in the UI and if enum strings are present it's a 2+ step search, especially if i18n usage isn't obvious), pretty harmless in this instance. This project is likely to have i18n I think, so I'm not super opposed, but I do want to be cautious of a ton of forward planning around i18n (I regret implementing i18n on https://udt.dev, because it created a slew of issues and made contribution pretty difficult, even though we saw a lot of adoption as a result I don't think it was worth the velocity).

I think if we could implement it such that english-first search is still effective e.g.

```
t("digits")
t("Device")
t("Status")
```

then always keep those inline. This is a bit easier because there's no redefining the english translation. The con is you have to repeat english for long strings, but this can be mitigated by supported a shorthand mapping e.g. `t("terms-of-service")` can still map to a large string

Reply:
Yea, another issue with the object is that unused keys are not linted against. This pseudo i18n has three nice features: first, it keeps the text out of the JSX which often makes the code harder to read / creates annoying diffs / makes typos harder to spot, etc; and second, the text is not that far away from where it is used since it stays in the same file alongside the component (which partially mitigates your primary gripe); third, it isolates the localized text so it may be factored out later.

Reply:
ok i'm down to try it!